File: problem/easy/pivotIndex.py
from typing import List
from wsgiref.util import request_uri


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)
        left_sum = 0
        for i in range(0,len(nums)):
            if left_sum == total - left_sum - nums[i]:
                return i
            left_sum += nums[i]
        return -1



test = Solution()
print(test.pivotIndex([1,7,3,6,5,6]))
print(test.pivotIndex([2,1,-1]))

File: problem/easy/pivotIndex.cpp
//
// Created by arnau on 26/05/2024.
//
#include <vector>
int pivotIndex(<vector<int>& nums) {
    int sum = 0, leftSum = 0;
    for (int num : nums) sum += num;
    for (int i = 0; i < nums.size(); i++) {
        if (leftSum == sum - leftSum - nums[i]) return i;
        leftSum += nums[i];
    }
    return -1;
}